File: app/src/main/java/com/github/dedis/student20_pop/model/network/layer/base/Unsubscribe.java
package com.github.dedis.student20_pop.model.network.layer.base;

/**
 * Unsubscribe from a channel
 */
public final class Unsubscribe extends Query {

    public Unsubscribe(String channel, int id) {
        super(channel, id);
    }

    @Override
    public String getMethod() {
        return Method.UNSUBSCRIBE.getMethod();
    }
}

File: be1-go/message/query/method/unsubscribe.go
package method

import "popstellar/message/query"

// Unsubscribe defines a JSON RPC unsubscribe message
type Unsubscribe struct {
	query.Base

	ID int `json:"id"`

	Params UnsubscribeParams `json:"params"`
}

type UnsubscribeParams struct {
	Channel string `json:"channel"`
}
